lib/tasks/sap_update_authorizations.rake

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+namespace :sap do
+  desc "Updates users metadata in authorizations by SAP"
+  task update_metadata: :environment do
+    users = Decidim::User.all
```

Comment:
I think this is still loading the array with all users in memory.
Let's remove it and directly use `Decidim::User.find_each` in the following line.